libr/search/search.c

```diff
@@ -427,7 +406,9 @@ R_API RList *r_search_find(RSearch *s, ut64 addr, const ut8 *buf, int len) {
 
 /* --- keywords --- */
 R_API int r_search_kw_add(RSearch *s, RSearchKeyword *kw) {
-	if (!kw) return false;
```

Comment:
I was really, really sad when I saw the following code for the first time:

```c
#define r_list_foreach(list, it, pos)\
	if (list)\
		for (it = list->head; it && (pos = it->data); it = it->n)
```

`pos = it->data` is dangerous, what if one of the elements is `NULL`? It will stop iterating at that element, ignoring the rest.

`if (list)` will you miss something if the for loop has not been executed? The (no-null-dereference) illusion comes with the cost that more trouble may accompany.